src/input/datetime.rs

```diff
@@ -508,11 +512,18 @@ pub struct TzInfo {
 #[pymethods]
 impl TzInfo {
     #[new]
-    fn new(seconds: i32) -> Self {
-        Self { seconds }
+    fn py_new(seconds: f32) -> PyResult<Self> {
```

Comment:
the change from `i32` to `f32` is to accepts the output of `timedelta.total_seconds()` directly

Reply:
Better to add

```rs
impl TryFrom<i32> for TzInfo {
    type Error = PyErr;

    fn try_from(seconds: i32) -> PyResult<Self> {
        if seconds.abs() >= 86400 {
            Err(PyValueError::new_err(format!(
                "TzInfo offset must be strictly between -86400 and 86400 (24 hours) seconds, got {seconds}"
            )))
        } else {
            Ok(Self { seconds })
        }
    }
}
```

Then use `(seconds.trunc() as i32).try_into()` here, then you can use `let tz_info: TzInfo = offset.try_into()?;` above and avoid going `i32 -> f32 -> i32`.

Reply:
nice